### DISCUSSION TITLE
Capcha Error setting keys

### DISCUSSION THREAD

**Discussion by jeromeatneotek** [givanz/Vvveb#85] [General]
`Internal server error!
(error 500)
pg_result_error(): Argument #1 ($result) must be of type PgSql\Result, false given

C:\Domains\myweb\wwwroot\system\db\pgsql.php on line 290
Code
error_log('pgsql error: ' . pg_result_error($this->last_res) . pg_last_error(self :: $link)); // <==

		} else {

			$this->last_res = pg_query(self :: $link, $sql);

			//pg_send_query(self :: $link, $sql);

			//$this->last_res = pg_get_result(self :: $link);

		}

		if ($this->last_res == false) {

			error_log('pgsql error: ' . pg_result_error($this->last_res) . pg_last_error(self :: $link));	 // <==

			error_log($this->debugSql($origSql, $params, $paramTypes));

		}

		return $this->last_res;

	}

Trace
#0 C:\Domains\myweb\wwwroot\system\db\pgsql.php(291): pg_result_error()
#1 C:\Domains\myweb\wwwroot\storage\model\admin\settingsql.pgsql.php(133): Vvveb\System\Db\Pgsql->execute()
#2 C:\Domains\myweb\wwwroot\system\setting.php(134): Vvveb\Sql\SettingSQL->set()
#3 C:\Domains\myweb\wwwroot\system\functions.php(116): Vvveb\System\Setting->multiSet()
#4 C:\Domains\myweb\wwwroot\plugins\captcha\admin\controller\settings.php(38): Vvveb\set_settings()
#5 [internal function]: Vvveb\Plugins\Captcha\Controller\Settings->save()
#6 C:\Domains\myweb\wwwroot\system\core\frontcontroller.php(229): call_user_func()
#7 C:\Domains\myweb\wwwroot\system\core\frontcontroller.php(298): Vvveb\System\Core\FrontController::call()
#8 C:\Domains\myweb\wwwroot\system\core\frontcontroller.php(364): Vvveb\System\Core\FrontController::redirect()
#9 C:\Domains\myweb\wwwroot\system\core\startup.php(355): Vvveb\System\Core\FrontController::dispatch()
#10 C:\Domains\myweb\wwwroot\index.php(138): Vvveb\System\Core\start()
#11 C:\Domains\myweb\wwwroot\admin\index.php(41): include('...')
#12 C:\Domains\myweb\wwwroot\public\admin\index.php(26): include('...')
#13 {main}`

**Comment by givanz** [comment]
Sorry for this, development is done mostly on mysql and pgsql was not tested with some recent changes.

This bug is fixed in the latest.zip you will need to redownload and update the files.

To reinstall just delete `config/db.php` and the installation page will show automatically, this will remove existing database tables and recreate them.

**Reply by jeromeatneotek** [reply]
I assume this will render any setup I have already done obsolete? This really isn't a solution for me moving forward. I am happy to assist with development and testing on the pgsql side if this helps. Is it possible to create database fixes for problems like this as opposed to a complete re-install?

**Reply by givanz** [reply]
To avoid reinstall you can manually run the sql from `/install/sql/pgsql/schema/site/setting.sql` and `setting_content.sql` to recreate the tables with the correct primary key.

These tables are used only for plugin settings and content will not be lost.

**Reply by jeromeatneotek** [reply]
I manually ran the two scripts suggested above, however now I do not see both my site options under admin in the data-v-component-sites drop down. I cannot see why this wouldn't display what is in the sites table in the database. I have cleared the cache. I only see the Default Site option, which would indicate that the sites are not being populated

**Reply by givanz** [reply]
Sorry forgot to mention to skip config folder on update.

Site configuration is stored in `config/sites.php` with some duplicate data that is stored in the database this is by design to load site configuration before opening database connection for caching purpose.

**Reply by jeromeatneotek** [reply]
OK, can I suggest it would be super helpful if there was a way of determining what is standard and what is config with regards to files, either by filename of extra sub-folder

**Reply by givanz** [reply]
The only data that is stored in both database and files is site configuration like name, host, theme, site settings is stored only in the database.

All other configuration data is not duplicated to database.

The reason to also use the database for site data is to generate the unique id for table relations like post_to_site, product_to_site etc and also have the data available for join queries to simplify code.

**Reply by jeromeatneotek** [reply]
OK I understand, however each time I grab the new zip file and load it to my website root, all the plugins are deactivated. It would be much better if any files that get updated via the admin site were placed in a separate folder so that they are not overwritten. Perhaps that is what the config folder is already? It looks like it is, can you confirm that one should not overwrite the files in the config folder, but one might need to add new files if additional configs are added?

**Reply by givanz** [reply]
This happens because plugin activation status is also stored in `config/plugins.php`

Having multiple files will make installation more complex, you only need to skip config folder and the theme folder `/public/themes/landing` if you make changes to theme files.

The zip is mostly used for new installations.

Automatic update properly skips config folder and only updates core files.

I don't use it to push updates yet because there are many updates and sometimes I can make slight database structure changes, I will use it after 1.0 is released.
